infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -172,6 +187,15 @@ private void batchWriteProcessor() {
                         currOp.setException(e);
                         res.add(currOp);
                     }
+                } else if (currOp.getType() == BatchWriterOperation.Type.BULK_WRITE) {
+                    try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)